src/sentry/rules/actions/utils.py

```diff
@@ -36,12 +36,36 @@ def generate_diff_labels(
     rule: Rule,
     changed_data: DefaultDict[str, list[str]],
     key: str,
-    statement: str,
 ) -> DefaultDict[str, list[str]]:
+    added_statement = "Added {} '{}'"
+    removed_statement = "Removed {} '{}'"
+    added_items = []
+
+    # Added items that are not seen in the prior rule data
     for data in prior_state.get(key, []):
         if data not in present_state.get(key, []):
             label = generate_rule_label(rule.project, rule, data)
-            changed_data[data["id"]].append(statement.format(label))
+            changed_data[data["id"]].append(added_statement.format(key[:-1], label))
+            added_items.append(data["id"])
+
+    # Check if the id is in both but the data has changed
+    for item in added_items:
+        if changed_data.get(item):
+            for data in prior_state.get(key, []):
```

Comment:
🤔 

is this going to be the same list for every `item in added_items`?
Looks like `key` and `prior_state` are being passed in?

afaict we're only ever utilizing `prior_state` by grabbing the key from it, so maybe we can construct the `prior_state_list` at the head of the function, or simply pass it in as an arg?